tensorrt_llm/_torch/pyexecutor/model_engine.py

```diff
@@ -1909,10 +1910,11 @@ def forward(self,
             return outputs
 
     def model_forward(self, **kwargs):
-        if is_trace_enabled("TLLM_TRACE_MODEL_FORWARD"):
-            return trace_func(self.model.forward)(**kwargs)
-        else:
-            return self.model.forward(**kwargs)
+        with torch.cuda.stream(self.stream):
```

Comment:
Considering decoder(sampler) is still on the default stream, I think we need careful stream synchronization between different streams.